src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -50,9 +50,18 @@
     </PropertyGroup>
 
     <PropertyGroup>
-      <LangVersion Condition="'$(LangVersion)' == '' AND
-        (('$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(TargetFrameworkVersion)' == 'v3.0') OR
-         ('$(TargetFrameworkIdentifier)' == '.NETStandard' AND '$(TargetFrameworkVersion)' == 'v2.1'))">preview</LangVersion>
+      <!-- To aid certain IDE features we make a distinction between the maximum supported language version, and the version 
+           actually used during compilation. .NETCore 3.0 supports up to C# 8.0, while all others support up to 7.3 by default.
+           Note that other frameworks (e.g., Xamarin, Unity, etc.) are expected to override this by setting
+           MaxSupportedLangVersion themselves. -->
+      <!-- Strip the 'v' off the beginning of TargetFrameworkVersion so we can actually compare it as a version. -->
+      <_TargetFrameworkVersionAsVersion>$(TargetFrameworkVersion.Substring(1))</_TargetFrameworkVersionAsVersion>
+      <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == '' AND
+                                          '$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND
+                                          '$(_TargetFrameworkVersionAsVersion)' >= '3.0'">8.0</MaxSupportedLangVersion>
```

Comment:
>8.0 [](start = 90, length = 3)

Requesting @agocke to take a look at this. This is going to quickly become a problem when someone forgets to update this targets file in the future, Andy is there any way we can tie this to the general language version infra we already have? Otherwise, we should add something to build boss to check this and be sure they're in sync.